src/components/__tests__/DynamicWidgets.js

```diff
@@ -0,0 +1,420 @@
+
+  localVue.use(plugin);
+
+  __setState({
+    attributesToRender: ['test2'],
+  });
+
+  const wrapper = mount(DynamicWidgets, {
+    localVue,
+    propsData: {
+      transformItems: items => items,
+    },
+    slots: {
+      default: `
+        <ais-menu attribute="test1" />
+        <ais-panel>
+          <ais-refinement-list attribute="test2" />
+        </ais-panel>
+      `,
+    },
+    stubs: {
+      'ais-refinement-list': MockRefinementList,
+      'ais-menu': MockMenu,
+    },
+  });
+
+  expect(wrapper.html()).toMatchInlineSnapshot(`
+
+<div class="ais-DynamicWidgets">
+  <div class="ais-DynamicWidgets-widget">
+    <div class="ais-Panel">
+      <div class="ais-Panel-body">
+        <div widget-name="ais-refinement-list"
+             attribute="test2"
+        >
+        </div>
+      </div>
+    </div>
+  </div>
+</div>
+
+`);
+});
+
+it('renders attributesToRender (hierarchical menu)', () => {
+  const localVue = createLocalVue();
+
+  localVue.use(plugin);
+
+  __setState({
+    attributesToRender: ['test1'],
+  });
+
+  const wrapper = mount(DynamicWidgets, {
+    localVue,
+    propsData: {
+      transformItems: items => items,
+    },
+    slots: {
+      default: `
+        <ais-hierarchical-menu :attributes="['test1','test2']" />
+        <ais-menu attribute="test3" />
+        <ais-panel>
+          <ais-refinement-list attribute="test4" />
+        </ais-panel>
+      `,
+    },
+    stubs: {
+      'ais-refinement-list': MockRefinementList,
+      'ais-menu': MockMenu,
+      'ais-hierarchical-menu': MockHierarchicalMenu,
+    },
+  });
+
+  expect(wrapper.html()).toMatchInlineSnapshot(`
+
+<div class="ais-DynamicWidgets">
+  <div class="ais-DynamicWidgets-widget">
+    <div widget-name="ais-hierarchical-menu"
+         attributes="test1,test2"
+    >
+    </div>
+  </div>
+</div>
+
+`);
+});
+
+it('updates DOM when attributesToRender changes', () => {
+  const localVue = createLocalVue();
+
+  localVue.use(plugin);
+
+  let attributesToRender = ['test1'];
+
+  __setState({
+    get attributesToRender() {
+      return attributesToRender;
+    },
+  });
+
+  const wrapper = mount(DynamicWidgets, {
+    localVue,
+    propsData: {
+      transformItems: items => items,
+    },
+    slots: {
+      default: `
+        <ais-hierarchical-menu :attributes="['test1','test2']" />
+        <ais-menu attribute="test3" />
+        <ais-panel>
+          <ais-refinement-list attribute="test4" />
+        </ais-panel>
+      `,
+    },
+    stubs: {
+      'ais-refinement-list': MockRefinementList,
+      'ais-menu': MockMenu,
+      'ais-hierarchical-menu': MockHierarchicalMenu,
+    },
+  });
+
+  expect(wrapper.html()).toMatchInlineSnapshot(`
+
+<div class="ais-DynamicWidgets">
+  <div class="ais-DynamicWidgets-widget">
+    <div widget-name="ais-hierarchical-menu"
+         attributes="test1,test2"
+    >
+    </div>
+  </div>
+</div>
+
+`);
+
+  attributesToRender = ['test3'];
+
+  wrapper.vm.$forceUpdate();
```

Comment:
can you try 

```js
await wrapper.setData({ attributesToRender: ['test3'] });
```

not sure if it works in this VTU version, though.

Reply:
doesn't seem to work, i think because of the mock

Reply:
doesn't seem to work, i think because of the mock